src/Time.cpp

```diff
@@ -31,6 +34,14 @@ Time Time::now()
     return Time(static_cast<int64_t>(t.tv_sec) * UsecPerSec + t.tv_usec);
 }
 
+static auto monotonicClock = std::chrono::steady_clock();
```

Comment:
Why don't you keep monotonicClock inside the function scope of Time::monotonic? It is not used anywhere else.

Reply:
Because then initialization would not be thread safe (clock.now() is thread safe in C++11)

Reply:
According to the standard 11 draft: http://www.open-std.org/JTC1/SC22/WG21/docs/papers/2011/n3242.pdf
6.7.4:

"If control enters the declaration concurrently while the variable is being initialized, the concurrent execution shall wait for
completion of the initialization."

(see also https://iamroman.org/blog/2017/04/cpp11-static-init/)

Related to this for gcc: 
```
 -fno-threadsafe-statics
           Do not emit the extra code to use the routines specified in the C++
           ABI for thread-safe initialization of local statics.  You can use
           this option to reduce code size slightly in code that doesn't need
           to be thread-safe.

Reply:
Cool. Did not know that C++11 fixed that. Will change.

Reply:
Fixed.